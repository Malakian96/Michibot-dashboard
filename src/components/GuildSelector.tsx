/* eslint-disable @typescript-eslint/no-explicit-any */
const GuildSelector = ({
  guilds,
  selectedGuild,
  setSelectedGuild,
  setChannels,
}: any) => {
  const handleGuildChange = (e: any) => {
    const guildId = e.target.value;
    setSelectedGuild(guildId);
    const selectedGuild = guilds.find(
      (guild: any) => guild.guildId === guildId
    );
    setChannels(selectedGuild ? selectedGuild.voiceChannels : []);
  };

  return (
    <select
      value={selectedGuild}
      onChange={handleGuildChange}
      className="bg-discord-dark-light text-white px-4 py-2 rounded-md mb-4"
    >
      <option value="">Select a server</option>
      {guilds.map((guild: any) => (
        <option key={guild.guildId} value={guild.guildId}>
          {guild.guildName}
        </option>
      ))}
    </select>
  );
};

export default GuildSelector;
