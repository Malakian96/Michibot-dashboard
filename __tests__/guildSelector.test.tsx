import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import GuildSelector from '../src/components/GuildSelector';

const mockGuilds = [
  {
    guildId: '1',
    guildName: 'Guild 1',
    voiceChannels: [
      { id: '1', name: 'Channel 1' },
      { id: '2', name: 'Channel 2' },
    ],
  },
  {
    guildId: '2',
    guildName: 'Guild 2',
    voiceChannels: [
      { id: '3', name: 'Channel 3' },
      { id: '4', name: 'Channel 4' },
    ],
  },
];

describe('GuildSelector', () => {
  it('renders correctly with guilds', () => {
    const { getByText } = render(
      <GuildSelector
        guilds={mockGuilds}
        selectedGuild=""
        setSelectedGuild={() => {}}
        setChannels={() => {}}
      />,
    );
    expect(getByText('Select a server')).toBeInTheDocument();
    expect(getByText('Guild 1')).toBeInTheDocument();
    expect(getByText('Guild 2')).toBeInTheDocument();
  });

  it('calls setSelectedGuild and setChannels on guild change', () => {
    const setSelectedGuild = jest.fn();
    const setChannels = jest.fn();

    const { getByRole } = render(
      <GuildSelector
        guilds={mockGuilds}
        selectedGuild=""
        setSelectedGuild={setSelectedGuild}
        setChannels={setChannels}
      />,
    );

    fireEvent.change(getByRole('combobox'), { target: { value: '1' } });

    expect(setSelectedGuild).toHaveBeenCalledWith('1');
    expect(setChannels).toHaveBeenCalledWith(mockGuilds[0].voiceChannels);
  });
});

describe('GuildSelectorSnapshot', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <GuildSelector
        guilds={mockGuilds}
        selectedGuild=""
        setSelectedGuild={() => {}}
        setChannels={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
