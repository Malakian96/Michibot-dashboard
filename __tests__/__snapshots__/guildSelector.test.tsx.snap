// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuildSelectorSnapshot matches snapshot 1`] = `
<div>
  <select
    aria-label="Select a server"
    class="bg-discord-dark-light text-white px-4 py-2 rounded-md mb-4"
  >
    <option
      value=""
    >
      Select a server
    </option>
    <option
      value="1"
    >
      Guild 1
    </option>
    <option
      value="2"
    >
      Guild 2
    </option>
  </select>
</div>
`;
